docs/quickstarts/cosign-akv-azure.md

```diff
@@ -0,0 +1,441 @@
+    ```
+
+    > In this article, use flag `--tlog-upload=false` to skip upload the signature to the transparent log (Rekor by default). 
+    > Sign using key in AKV does not necessarily require the role `Key Vault Crypto Officer`, you can login with another identity and assign the role `Key Vault Crypto User` for signing action.
+
+    Confirm the signature is pushed and associated with the image in ACR
+
+    ```bash
+    cosign tree $IMAGE_SIGNED
+    ```
+
+## Validate signatures at K8s admission control
+
+### Access control
+
+User assigned managed identity is used for Ratify to access ACR and AKV. 
+
+1. Configure environment variables
+
+```bash
+export ID_NAME=<your managed id>
+export ID_RG=<your managed id resource group>
+```
+
+1. Create a user assigned managed identity
+
+    ```bash
+    az identity create --name $ID_NAME --resource-group $ID_RG
+    export OBJECT_ID=$(az identity show --name $ID_NAME --resource-group $ID_RG --query 'principalId' -o tsv)
+    ```
+
+1. Assign `acrpull` role to this identity for accessing ACR
+
+    ```bash
+    az role assignment create --assignee-object-id $OBJECT_ID --role acrpull \
+    --scope "/subscriptions/$SUB_ID/resourceGroups/$ACR_RG/providers/Microsoft.ContainerRegistry/registries/$ACR_NAME"
+    ```
+
+1. Assign `Key Vault Crypto User` role to this identity for accessing AKV
+
+    ```bash
+    az role assignment create --role "Key Vault Crypto User" --assignee $OBJECT_ID \
+    --scope "/subscriptions/$SUB_ID/resourceGroups/$AKV_RG/providers/Microsoft.KeyVault/vaults/$AKV_NAME"
+    ```
+
+### Set up your AKS cluster
+
+1. Configure environment variables
+
+    ```bash
+    export CLIENT_ID=$(az identity show --name  $ID_NAME --resource-group $ID_RG --query 'clientId' -o tsv)
+    export TENANT_ID=$(az identity show --name $ID_NAME --resource-group $ID_RG --query tenantId -o tsv)
+    ```
+
+1. Update the AKS cluster with OIDC and workload identity support
+
+    ```bash
+    az aks update -g $AKS_RG -n $AKS_NAME --enable-oidc-issuer --enable-workload-identity
+    ```
+
+1. Create federate credential for your managed identity
+
+    ```bash
+    export AKS_OIDC_ISSUER="$(az aks show -n $AKS_NAME -g $AKS_RG --query "oidcIssuerProfile.issuerUrl" -o tsv)"
+
+    az identity federated-credential create --name ratify-federated-credential --identity-name $ID_NAME -g $ID_RG --issuer $AKS_OIDC_ISSUER --subject system:serviceaccount:"gatekeeper-system":"ratify-admin"
+    ```
+
+### Install Ratify
+
+1. AKS with Azure policy enabled
+   
+   Make sure AKS image integrity policy is not assigned to your AKS cluster.
+
+   ```bash
+   helm install ratify oci://ghcr.io/deislabs/ratify-chart-dev/ratify --atomic \
+   --version v1.2.0 \
+   --namespace gatekeeper-system \
+   --set azureWorkloadIdentity.clientId="$CLIENT_ID"
+   --set featureFlags.RATIFY_CERT_ROTATION=true \
+   --set image.pullPolicy=Always \
```

Comment:
If a a cosign cert is specified, the charts are wired up to apply the cosign verifer.  We can install Ratify with cosign enabled, but use kubectl describe command to walk through the verifier and KMP CRs deployed.

helm install ratify \
./charts/ratify --atomic \
    --namespace gatekeeper-system \
    --set-file cosignKeys[0]=/home/azureuser/cosignTestMay24/2048/import-cosign.pub \
    --set featureFlags.RATIFY_CERT_ROTATION=true

Reply:
I would like to use configuration yaml later for configuring Cosign or Notation verifier, because during installation user may not know what exactly the configure, and there could be some configuration errors, which is not easy to detect during installation. If you apply the configuration via custom resource, it's easy to tell user how to check the results.